index.js

```diff
@@ -7,14 +7,17 @@ const LINE_OFFSET = 7
 const { symbols } = require('pino')
 const { asJsonSym } = symbols
 
-function traceCaller (pinoInstance) {
+function traceCaller (pinoInstance, relativeTo) {
   function get (target, name) {
     return name === asJsonSym ? asJson : target[name]
   }
 
   function asJson (...args) {
     args[0] = args[0] || Object.create(null)
     args[0].caller = Error().stack.split('\n').slice(2).filter(s => !s.includes('node_modules/pino') && !s.includes('node_modules\\pino'))[STACKTRACE_OFFSET].substr(LINE_OFFSET)
+    if (typeof relativeTo === 'string') {
+      args[0].caller = args[0].caller.replace(relativeTo + '/', '').replace(relativeTo + '\\', '')
```

Comment:
please use https://nodejs.org/api/path.html#path_path_relative_from_to instead

Reply:
I was considering it but decided against because it requires reliably parsing filepath from stacktrace message. I see two options to achieve that:

1. I could try to do that myself, but:
-  I'm not sure if I will find all possible kinds of stacktraces so every case is covered by my parsing
-  you will have to maintain it in case stacktrace format changes
2. I could use some maintained library for parsing stacktraces, e.g. https://github.com/xpl/stacktracey:
- all the drawbacks for adding a dependency to a super lightweight project
- however the current implementation will become much clearer and more stable
- also I will be able to do even more by assigning all parts of stacktrace to different members of log object so user will be able to format it as he wants: `{..., "caller_function": "Object.<anonymous>", "path": "index.js", "line": 8, "column": 12, ...}`

What do you think?

Reply:
.replace() is totally ok!

Reply:
Do you mean .replace() that I already implemented or something else? How can I proceed?

Reply:
This is ok.